--- src/rfs_inode.c.orig	2010-06-18 15:44:13.000000000 +0200
+++ src/rfs_inode.c	2012-01-10 18:18:16.000000000 +0100
@@ -742,7 +742,9 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,34)
 	if ((iattr->ia_valid & ATTR_UID && iattr->ia_uid != inode->i_uid) ||
 	    (iattr->ia_valid & ATTR_GID && iattr->ia_gid != inode->i_gid))
-		return rfs_dq_transfer(inode, iattr) ? -EDQUOT : 0;
+		rv = rfs_dq_transfer(inode, iattr);
+	if (rv)
+		return -EDQUOT;
 #endif
 
 	return inode_setattr(inode, iattr);
